fix(config): Adjust MSRV resolve config field name / values (#14296)
rust-lang/cargo: src/doc/src/reference/unstable.md

### MSRV-aware resolver

`-Zmsrv-policy` allows access to an MSRV-aware resolver which can be enabled with:
- `resolver.incompatible-rust-versions` config field
- `workspace.resolver = "3"` / `package.resolver = "3"`
- `package.edition = "2024"` (only in workspace root)

The resolver will prefer dependencies with a `package.rust-version` that is the same or older than your project's MSRV.
Your project's MSRV is determined by taking the lowest `package.rust-version` set among your workspace members.

> If there is none set, your toolchain version will be used with the intent to pick up the version from rustup's `rust-toolchain.toml`, if present.

Reviewer: Slightly off-topic, but we may not want to expose the implementation detail of rustup.

```suggestion
If there is none set, your toolchain version will be used. The toolchain version is usually determined by `rustup` or the system package manager. 
```

Author: This section isn't as important long-term as this is just describing the unstable feature.

However, in describing it this way, I'm trying to accomplish two things
- Emphasize the motivating use case for this functionality
- Help rustup users discover this by using terms they would expect

I am making a slight tweak to it so hopefully that will help.

#### `resolver.incompatible-rust-versions`
* Type: string
* Default: "allow"
* Environment: `CARGO_RESOLVER_INCOMPATIBLE_RUST_VERSIONS`

> Select how packages with incompatible rust-versions should be resolved.  Values include:

Reviewer: It is unclear what an incompatible rust-version is. Should we link "incompatible rust-versions" to `package.rust-version` in the manifest?

Author: I've called out the field name.  We should eventually turn all of those into links.

> - `allow`: treat them like any other dependency

Reviewer: This is also unclear as user doesn't know how other dependency is treated. Are we making it vague for rooms for future behavior changes?

Author: I can't think of a reason to be intentionally vague about `allow`, I'm just unsure of what more needs to be described.

Reviewer: I think the problem is pronouns and “other”. What is included in “other”? Maybe something like

```suggestion
- `allow`: treat `package.rust-version` incompatible dependency as compatible
- `fallback`: only consider `package.rust-version` incompatible dependency if no other compatible dependency version matched
```

Pronouns are often confusing, and I prefer to use them less.

(Maybe it is just me 😬)

Author: To be fully correct, its an "incompatible dependency version" (not just dependency).  This starts to become a mouthful.  I gave it a try by skipping on `package.` and just saying "version".

- `fallback`: only consider them if no other dependency version matched

Can be overridden with
- `--ignore-rust-version` CLI option

> - Setting the dependency's version requirement above any version that is rust-version compatible

Reviewer: Should we highlight the field name?

```suggestion
- Setting the dependency's version requirement above any version with a compatible `package.rust-version`
```
